### PR TITLE
Support cgroup namespaces for cgroup v1

### PR DESCRIPTION
**Pull request by Furisto**
This adds support for namespacing v1 control cgroups. 

Tasks still open
- Emulate namespacing behavior even if no cgroup namespace is specified
- Add support for cgroup v2

I will add this in another PR.

### CODE REVIEW

**Comment by utam0k** [comment]
Perhaps this is a good feature. Can I ask you to open the issue about this and describe more details?

**Comment by Furisto** [comment]
#332 is the feature 😉

**Comment by utam0k** [comment]
@Furisto Thanks! That's a great description. I understand.

File: cgroups/src/v1/controller_type.rs [hunk 1]
```diff
@@ -37,6 +37,25 @@ impl Display for ControllerType {
     }
 }
 
+impl AsRef<str> for ControllerType {
```
Comment (utam0k) [inline cgroups/src/v1/controller_type.rs:40]: 👍

File: src/process/init.rs [hunk 2]
```diff
@@ -187,11 +187,11 @@ pub fn container_init(
                 && ns_type != CloneFlags::CLONE_NEWPID
                 && ns_type != CloneFlags::CLONE_NEWNS
         })
-        .with_context(|| "Failed to apply namespaces")?;
+        .with_context(|| "failed to apply namespaces")?;
```
Comment (utam0k) [inline src/process/init.rs:190]: 👍

File: src/process/intermediate.rs [hunk 3]
```diff
@@ -60,6 +63,17 @@ pub fn container_intermediate(
             .with_context(|| format!("Failed to enter pid namespace: {:?}", pid_namespace))?;
     }
 
+    // this needs to be done before we create the init process, so that the init
+    // process will already be captured by the cgroup
+    if args.rootless.is_none() {
+        apply_cgroups(
```
Comment (utam0k) [inline src/process/intermediate.rs:69]: Can I ask you to update  [docs/.drawio.svg](https://github.com/containers/youki/blob/main/docs/.drawio.svg)  using  [vscode-drawing](https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio).

**Comment by Furisto** [comment]
@utam0k Done

**Review by utam0k — APPROVED** [review]
Looks good!

**Pull request merged**